File: src/main/java/it/astromark/agenda/schoolclass/dto/SignHourRequest.java
package it.astromark.agenda.schoolclass.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import it.astromark.classwork.dto.ClassActivityRequest;
import it.astromark.classwork.dto.HomeworkRequest;

import java.time.LocalDate;

@Schema(description = "Request object for signing a teaching hour")
public record SignHourRequest(
        Integer id,
        Integer hour,
        String subject,
        LocalDate date,
        ClassActivityRequest activity,
        HomeworkRequest homework
) {
}

File: src/main/typescript/src/entities/SignHourRequest.ts
export type SignHourRequest = {
    id: number | null;
    hour: number | null;
    subject: string | null;
    date: Date;
    activity: ClassActivityRequest | null;
    homework: HomeworkRequest | null;
}

export type ClassActivityRequest = {
    id: number | null;
    title: string;
    description: string;
}

export type HomeworkRequest = {
    id: number | null;
    title: string;
    description: string;
    dueDate: Date;
    hasChat: boolean;
}